lib/Dialect/FIRRTL/Transforms/ModuleInliner.cpp

```diff
@@ -934,309 +940,451 @@ bool Inliner::shouldInline(Operation *op) {
+        if (toBeFlattened) {
+          if (failed(flattenInto(nestedPrefix, childIL, mapper, {})))
+            return WalkResult::interrupt();
+        } else {
+          // Recursively inline all the child modules under `parent`, that are
+          // marked to be inlined.
+          if (failed(inlineInto(nestedPrefix, childIL, mapper, symbolRenames)))
+            return WalkResult::interrupt();
+        }
+        currentPath.pop_back();
+        activeHierpaths = parentActivePaths;
 
-    // The InstanceOp `instance` might not have a symbol, if it does not
-    // participate in any HierPathOp. But the reTop might add a symbol to it, if
-    // a HierPathOp is added to this Op.
-    DenseMap<Attribute, Attribute> symbolRenames;
-    if (!rootMap[target.getNameAttr()].empty() && !toBeFlattened) {
-      for (auto sym : rootMap[target.getNameAttr()]) {
-        auto &mnla = nlaMap[sym];
-        sym = mnla.reTop(module);
-        StringAttr instSym = getOrAddInnerSym(
-            instance, [&](FModuleLike mod) -> hw::InnerSymbolNamespace & {
-              return mic.modNamespace;
-            });
-        instOpHierPaths[InnerRefAttr::get(moduleName, instSym)].push_back(
-            cast<StringAttr>(sym));
-        // TODO: Update any symbol renames which need to be used by the next
-        // call of inlineInto.  This will then check each instance and rename
-        // any symbols appropriately for that instance.
-        symbolRenames.insert({mnla.getNLA().getNameAttr(), sym});
-      }
-    }
-    auto instInnerSym = getInnerSymName(instance);
-    auto parentActivePaths = activeHierpaths;
-    setActiveHierPaths(moduleName, instInnerSym);
-    // This must be done after the reTop, since it might introduce an innerSym.
-    currentPath.emplace_back(moduleName, instInnerSym);
-    // Create the wire mapping for results + ports. We RAUW the results instead
-    // of mapping them.
-    IRMapping mapper;
-    mic.b.setInsertionPoint(instance);
-    auto nestedPrefix = (instance.getName() + "_").str();
-
-    InliningLevel childIL(mic, target);
-    createDebugScope(childIL, instance);
-
-    mapPortsToWires(nestedPrefix, childIL, mapper, {});
-    for (unsigned i = 0, e = instance.getNumResults(); i < e; ++i)
-      instance.getResult(i).replaceAllUsesWith(childIL.wires[i]);
-
-    // Inline the module, it can be marked as flatten and inline.
-    if (toBeFlattened) {
-      flattenInto(nestedPrefix, childIL, mapper, {});
-    } else {
-      // Recursively inline all the child modules under `parent`, that are
-      // marked to be inlined.
-      inlineInto(nestedPrefix, childIL, mapper, symbolRenames);
-    }
-    currentPath.pop_back();
-    activeHierpaths = parentActivePaths;
+        // Erase the replaced instance.
+        instance.erase();
+        return WalkResult::skip();
+      });
 
-    // Erase the replaced instance.
-    instance.erase();
-  }
+  return failure(result.wasInterrupted());
```

Comment:
I did not review this in detail as it appears to also be code motion.

Reply:
Yep, split that into separate commit to help re:reviewing.